### PR TITLE
Add Analytics support

### PR DESCRIPTION
**Pull request by anxolin**
Adds Google analytics support

* Integrates `react-ga` library
* Creates a new API for exposing the analytics api
* The API exposes one init method, that would initiate the analytics
* The init method, also register any error
* Create a component `RouteAnalytics` to register every page change

### CODE REVIEW

File: src/apps/explorer/ExplorerApp.tsx [hunk 1]
```diff
@@ -11,6 +11,12 @@ import { GenericLayout } from 'components/layout'
 import { Header } from './layout/Header'
 
 import { NetworkUpdater } from 'state/network'
+import { initAnalytics } from 'api/analytics'
+import RouteAnalytics from 'components/common/RouteAnalytics'
+
+// Init analytics
+const GOOGLE_ANALYTICS_ID: string | undefined = process.env.GOOGLE_ANALYTICS_ID
+initAnalytics({ trackingCode: GOOGLE_ANALYTICS_ID || 'UA-190948266-2' })
```
Comment (alfetopito) [inline src/apps/explorer/ExplorerApp.tsx:19]: Is it ok to expose this?

**Review by alfetopito — APPROVED** [review]

File: src/api/analytics/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export function initAnalytics(params: InitAnalyticsParams): void {
+  const { trackingCode, options } = params
+  if (typeof trackingCode === 'string') {
+    ReactGA.initialize(trackingCode, options)
+
+    ReactGA.set({
+      customBrowserType: getCustomBrowserType(),
+    })
+  } else {
```
Comment (W3stside) [inline src/api/analytics/index.tsx:26]: ? what is this for

File: src/apps/explorer/ExplorerApp.tsx (hunk 1 above)
Comment (W3stside) [reply src/apps/explorer/ExplorerApp.tsx:19]: seconding

File: src/api/analytics/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import ReactGA, { InitializeOptions } from 'react-ga'
+import { isMobile } from 'react-device-detect'
+export { pageview as viewPage, event as triggerEvent } from 'react-ga'
+
+function getCustomBrowserType(): string {
+  if (isMobile) {
+    return 'ethereum' in window ? 'mobileWeb3' : 'mobileRegular'
```
Comment (W3stside) [inline src/api/analytics/index.tsx:7]: i don't like using the `in` operator, see why:
```js
window.ethereum = { a: 'b;ah' }
// { a: "blah" }

if ('ethereum' in window) console.log('yay')
"yay" // logs

// set ethereum to `undefined`
window.ethereum = undefined
if ('ethereum' in window) console.log('yay')
"yay" // logs

if (window.ethereum) console.log('yay')
undefined // no log
```

**Review by W3stside — CHANGES_REQUESTED** [review]
the `in` operator should be changed (`analytics/index.tsx`)

File: src/apps/explorer/ExplorerApp.tsx (hunk 1 above)
Comment (anxolin) [reply src/apps/explorer/ExplorerApp.tsx:19]: what's the concern?

File: src/api/analytics/index.tsx (hunk 3 above)
Comment (anxolin) [reply src/api/analytics/index.tsx:7]: okok

File: src/api/analytics/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/api/analytics/index.tsx:50]: This is a new dimension. To segment by the 4 values you see in the function implementation. Got the inspiration from uni code. Not an expert in analytics

**Review by W3stside — APPROVED** [review]

**Pull request merged**